LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ClientCertificateValidatorService.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan.NetworkServer.BasicsStation
+{
+    using System;
+    using System.Linq;
+    using System.Net.Security;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+
+    internal class ClientCertificateValidatorService
+    {
+        private readonly IBasicsStationConfigurationService stationConfigurationService;
+        private readonly ILogger<ClientCertificateValidatorService> logger;
+
+        public ClientCertificateValidatorService(IBasicsStationConfigurationService stationConfigurationService,
+                                                 ILogger<ClientCertificateValidatorService> logger)
+        {
+            this.stationConfigurationService = stationConfigurationService;
+            this.logger = logger;
+        }
+
+        // Following synchronous implementation is needed for ASP.NET Core ClientCertificateValidation
+        public virtual bool Validate(X509Certificate2 certificate, X509Chain? chain, SslPolicyErrors sslPolicyErrors) =>
```

Comment:
[nit] I would argue that we can do this on the call site instead of introducing an overload for this. This is not something we should "encourage" to reuse in other places